src/node/internal/zlib.d.ts

```diff
@@ -1 +1,115 @@
+export const Z_FIXED: number;
+export const Z_DEFAULT_STRATEGY: number;
+export const ZLIB_VERNUM: number;
+
+export const DEFLATE: number;
+export const INFLATE: number;
+export const GZIP: number;
+export const GUNZIP: number;
+export const DEFLATERAW: number;
+export const INFLATERAW: number;
+export const UNZIP: number;
+export const BROTLI_DECODE: number;
+export const BROTLI_ENCODE: number;
+
+export const Z_MIN_WINDOWBITS: number;
+export const Z_MAX_WINDOWBITS: number;
+export const Z_DEFAULT_WINDOWBITS: number;
+export const Z_MIN_CHUNK: number;
+export const Z_MAX_CHUNK: number;
+export const Z_DEFAULT_CHUNK: number;
+export const Z_MIN_MEMLEVEL: number;
+export const Z_MAX_MEMLEVEL: number;
+export const Z_DEFAULT_MEMLEVEL: number;
+export const Z_MIN_LEVEL: number;
+export const Z_MAX_LEVEL: number;
+export const Z_DEFAULT_LEVEL: number;
+
+export const BROTLI_OPERATION_PROCESS: number;
+export const BROTLI_OPERATION_FLUSH: number;
+export const BROTLI_OPERATION_FINISH: number;
+export const BROTLI_OPERATION_EMIT_METADATA: number;
+export const BROTLI_PARAM_MODE: number;
+export const BROTLI_MODE_GENERIC: number;
+export const BROTLI_MODE_TEXT: number;
+export const BROTLI_MODE_FONT: number;
+export const BROTLI_DEFAULT_MODE: number;
+export const BROTLI_PARAM_QUALITY: number;
+export const BROTLI_MIN_QUALITY: number;
+export const BROTLI_MAX_QUALITY: number;
+export const BROTLI_DEFAULT_QUALITY: number;
+export const BROTLI_PARAM_LGWIN: number;
+export const BROTLI_MIN_WINDOW_BITS: number;
+export const BROTLI_MAX_WINDOW_BITS: number;
+export const BROTLI_LARGE_MAX_WINDOW_BITS: number;
+export const BROTLI_DEFAULT_WINDOW: number;
+export const BROTLI_PARAM_LGBLOCK: number;
+export const BROTLI_MIN_INPUT_BLOCK_BITS: number;
+export const BROTLI_MAX_INPUT_BLOCK_BITS: number;
+export const BROTLI_PARAM_DISABLE_LITERAL_CONTEXT_MODELING: number;
+export const BROTLI_PARAM_SIZE_HINT: number;
+export const BROTLI_PARAM_LARGE_WINDOW: number;
+export const BROTLI_PARAM_NPOSTFIX: number;
+export const BROTLI_PARAM_NDIRECT: number;
+export const BROTLI_DECODER_RESULT_ERROR: number;
+export const BROTLI_DECODER_RESULT_SUCCESS: number;
+export const BROTLI_DECODER_RESULT_NEEDS_MORE_INPUT: number;
+export const BROTLI_DECODER_RESULT_NEEDS_MORE_OUTPUT: number;
+export const BROTLI_DECODER_PARAM_DISABLE_RING_BUFFER_REALLOCATION: number;
+export const BROTLI_DECODER_PARAM_LARGE_WINDOW: number;
+export const BROTLI_DECODER_NO_ERROR: number;
+export const BROTLI_DECODER_SUCCESS: number;
+export const BROTLI_DECODER_NEEDS_MORE_INPUT: number;
+export const BROTLI_DECODER_NEEDS_MORE_OUTPUT: number;
+export const BROTLI_DECODER_ERROR_FORMAT_EXUBERANT_NIBBLE: number;
+export const BROTLI_DECODER_ERROR_FORMAT_RESERVED: number;
+export const BROTLI_DECODER_ERROR_FORMAT_EXUBERANT_META_NIBBLE: number;
+export const BROTLI_DECODER_ERROR_FORMAT_SIMPLE_HUFFMAN_ALPHABET: number;
+export const BROTLI_DECODER_ERROR_FORMAT_SIMPLE_HUFFMAN_SAME: number;
```

Comment:
Do we really need all of these defines? I can't find a reference to this define in the node:zlib spec.

Reply:
I'm not sure - they're definitely all present when you import `node:zlib` but it's true that not all are documented.

Reply:
Maybe we can limit it to the documented ones https://nodejs.org/api/zlib.html#brotli-constants – it's probably not overly consequential, but we won't be able to remove constants later on based on the compatibility limit and everything we add to the JSG API increases the compile time of the main translation unit that includes all JSG definitions (workerd-api.c++ for workerd), which is a bottleneck for us.

Reply:
They're in Node's type definitions for zlib FWIW. 
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/ec8ec925cf490ce8825692cc2ea67c6cefd6d996/types/node/zlib.d.ts#L346

Reply:
I would keep them. In Node.js the actual exported values are the source of truth and not the documentation. It is possible that someone just forgot to document these or any number of things.